pkg/rkenodeconfigserver/nodeserver.go

```diff
@@ -344,13 +346,15 @@ func (n *RKENodeConfigServer) getServiceOptions(k8sVersion string) (map[string]i
 	if svcOptions != nil {
 		data["k8s-service-options"] = svcOptions
 	}
-	svcOptionsWindows, err := kd.GetRKEK8sServiceOptions(k8sVersion, n.serviceOptionsLister, n.serviceOptions, n.sysImagesLister, n.sysImages, kd.Windows)
-	if err != nil {
-		logrus.Errorf("getK8sServiceOptionsWindows: k8sVersion %s [%v]", k8sVersion, err)
-		return data, err
-	}
-	if svcOptionsWindows != nil {
-		data["k8s-windows-service-options"] = svcOptionsWindows
+	if osType == "windows" {
+		svcOptionsWindows, err := kd.GetRKEK8sServiceOptions(k8sVersion, n.serviceOptionsLister, n.serviceOptions, n.sysImagesLister, n.sysImages, kd.Windows)
+		if err != nil {
+			logrus.Errorf("getK8sServiceOptionsWindows: k8sVersion %s [%v]", k8sVersion, err)
+			return data, err
+		}
+		if svcOptionsWindows != nil {
+			data["k8s-windows-service-options"] = svcOptionsWindows
+		}
```

Comment:
What happens if `svcOptionsWindows` is nil?

Reply:
service options gets sent to GeneratePlan() which uses it in two places, both check if it's nil/not nil and handle it:
https://github.com/rancher/rancher/blob/master/vendor/github.com/rancher/rke/cluster/plan.go#L63-L66
https://github.com/rancher/rancher/blob/master/vendor/github.com/rancher/rke/cluster/plan.go#L999